location-search/SearchClient.test.ts

```diff
@@ -0,0 +1,141 @@
+          longitude: -122.34014899999994
+        },
+        placemark: {
+          name: "Amazon Go",
+          country: "USA",
+          postalCode: "98121",
+          street: "7th Ave",
+          streetNumber: "2131",
+          region: "Washington",
+          isoCountryCode: "US",
+          city: "Seattle"
+        }
+      })
+    })
+  })
+  describe("searchRecentLocations tests", () => {
+    it("gives a LocationSearchResult array when called with information in async storage", async () => {
+      const location1 = mockTiFLocation()
+      const location2 = mockTiFLocation()
+      const location3 = mockTiFLocation()
+      await asyncStorageSaveRecentLocation(location1, "attended-event")
+      await asyncStorageSaveRecentLocation(location2, "hosted-event")
+      await asyncStorageSaveRecentLocation(location3, "hosted-event")
+
+      const results = await searchRecentLocations()
+
+      expect(results).toEqual([
+        {
+          location: location3,
+          annotation: "hosted-event",
+          isRecentLocation: true
+        },
+        {
+          location: location2,
+          annotation: "hosted-event",
+          isRecentLocation: true
+        },
+        {
+          location: location1,
+          annotation: "attended-event",
+          isRecentLocation: true
+        }
+      ])
+    })
+    it("returns an empty array when called with faulty or nonexistent information", async () => {
+      const results = await searchRecentLocations()
+      expect(results).toEqual([])
+    })
+  })
+
+  describe("searchWithRecentAnnotations tests", () => {
+    it("returns an array of valid LocationSearchResults when given a proper callback", async () => {
+      const mockLocation1 = mockLocationCoordinate2D()
+      const mockLocation2 = mockLocationCoordinate2D()
+      const mockLocation3 = mockLocationCoordinate2D()
+      const mockTiFLocation1 = {
+        coordinates: mockLocation1,
+        placemark: { name: "New York" }
+      }
+      const mockTiFLocation2 = {
+        coordinates: mockLocation2,
+        placemark: { name: "Joe" }
+      }
+      const mockTiFLocation3 = {
+        coordinates: mockLocation3,
+        placemark: { name: "New York" }
+      }
+
+      const testQuery = new LocationsSearchQuery("New York")
+
+      await asyncStorageSaveRecentLocation(mockTiFLocation1, "attended-event")
+      await asyncStorageSaveRecentLocation(mockTiFLocation2, "hosted-event")
+      await asyncStorageSaveRecentLocation(mockTiFLocation3, "hosted-event")
+
+      const testResults = await searchWithRecentAnnotations(
+        testQuery,
+        mockLocationSearchFunction
+      )
+      expect(testResults).toEqual([
+        {
+          location: mockTiFLocation3,
+          annotation: "hosted-event",
+          isRecentLocation: true
+        },
+        {
+          location: mockTiFLocation1,
+          annotation: "attended-event",
+          isRecentLocation: true
+        }
+      ])
+    })
```

Comment:
I think you can add 2 more results to this assertion:
1. A result without an annotation but is a recent location.
2. A result that isn't a recent location and thus has no annotation.